cmd/nerdctl/client.go

```diff
@@ -24,18 +24,20 @@ import (
 	"strings"
 
 	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/namespaces"
 	"github.com/opencontainers/go-digest"
-	"github.com/urfave/cli/v2"
 )
 
-func newClient(clicontext *cli.Context, opts ...containerd.ClientOpt) (*containerd.Client, context.Context, context.CancelFunc, error) {
-	ctx := context.Background()
-	namespace := clicontext.String("namespace")
+func newClient(cmd *cobra.Command, opts ...containerd.ClientOpt) (*containerd.Client, context.Context, context.CancelFunc, error) {
+	ctx := cmd.Context()
+	namespace := viper.GetString("namespace")
```

Comment:
`viper.GetString("namespace")` means that get from flags if user explicitly specified, or get from envvar `CONTAINERD_NAMESPACE`, otherwise get default value.